src/suggestion-builder.js

```diff
@@ -0,0 +1,66 @@
+import _ from 'lodash'
+import { buildClause } from './utils'
+
+export default function suggestionBuilder(newSuggestion) {
+    let suggestions = _.isEmpty(newSuggestion) ? {} : newSuggestion
+
+    function makeSuggestion(type, field, options) {
+        let suggestName
+        const { name, text } = options
+
+        if (name) {
+            suggestName = name
+            _.unset(options, 'name')
+        } else {
+            suggestName = `suggest_${type}_${field}`
+        }
+
+        if (text) {
+            _.unset(options, 'text')
+        }
+
+        const innerClause = Object.assign({}, {
+            text,
+            [type]: buildClause(field, null, options)
+        })
+
+        Object.assign(suggestions, {
+            [suggestName]: innerClause
```

Comment:
My assumption of this comment is that this PR is only trying to make the basic suggester which is [Term Suggester](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-suggesters.html#term-suggester). In this case, here we should be comping an object, something like this:
```
        Object.assign(suggestions, {
            suggest: {
                [suggestName]: innerClause
            }
        })
```
The innerClause here should also be limited to `term` clause for this to work. It would be nice to pass multiple suggestions (different names and term clauses) which will all under `suggest` object.

Reply:
Yeah going to add support for completion, phrase, and context suggestions once we get the base locked down!